Fix indicator label circle shape for a 14px font-size label.

notlee opened the pull request:
edit: see https://github.com/Financial-Times/o-labels/pull/104
_____________________

Indicator labels such as the live label have a circle.
At some font sizes the width/height of the circle is a decimal.
For small font sizes, i.e 14px, the browser renders the circle
squashed, even though the width/height are equal. Setting the
transform property changes the way the browser renders the circle,
in a way which overcomes this problem

before, 14px font-size
![Screenshot 2020-11-11 at 10 53 45](https://user-images.githubusercontent.com/10405691/98803435-a74d4a80-240c-11eb-9e01-1dc18c891df8.png)

after, 14px font-size
![Screenshot 2020-11-11 at 10 53 41](https://user-images.githubusercontent.com/10405691/98803436-a7e5e100-240c-11eb-974d-34a8a84ea627.png)

notlee committed:
Fix indicator label circle shape for a 14px font-size label.

Indicator labels such as the live label have a circle.
At some font sizes the width/height of the circle is a decimal.
For small font sizes, i.e 14px, the browser renders the circle
squashed, even though the width/height are equal. Setting the
transform property changes the way the browser renders the circle,
in a way which overcomes this problem

origamiserviceuser commented:
o-labels bundle size difference from 5.1.4 to 2f5a98879a95d9b9822962e0e310b904bc5d8c55
No significant bundle size differences found.

chee commented on src/scss/_placeholders.scss:
```diff
@@ -32,6 +32,12 @@ $_o-labels-first-import: true !default;
 		left: 0;
 		width: calc(#{$_o-labels-indicator-circle-size-calc});
 		height: calc(#{$_o-labels-indicator-circle-size-calc});
+		// At some font sizes the width/height of the circle is a decimal.
+		// For small font sizes, i.e 14px, the browser renders the circle
+		// squashed, even though the width/height are equal. Setting the
+		// transform property changes the way the browser renders the circle,
+		// in a way which overcomes this problem
+		transform: scale(1);
```
can we have a little more information about what "changes the way the browser renders the circle" means? perhaps a link to the relevant part of the specification or browser docs?

does setting `opacity: 1` also work?

notlee replied:
Honestly I'm not sure. I think it might be to do with offloading to the GPU, which maybe handles subpixel rendering differently. Setting `opacity: 1` does nothing but setting `will-change: transform` has the same effect

notlee replied:
I guess a less hacky solution would be to make the indicator labels less flexible and have different classes which set whole values. `o-labels-indicator--xs`, `o-labels-indicator--s`, `o-labels-indicator--m`.

chee replied:
okay, so in Blink we're pushing it onto a compositing layer so that it's handled by the GPU and does hardware accelerated subpixel rendering.
i think that `will-change: opacity` would also make it work then, but the behaviours are documented in the rendering engines rather than specifications.
mostly this is browser internals stuff, but there is a little written about it in the [spec for will-change](https://drafts.csswg.org/css-will-change/#will-change).
i do prefer the idea of having classes that set whole values, but i'm going to hit approve on this pr.

notlee replied:
Looks like this still doesn't work in Safari/Webkit. Change of plan, let's avoid whole numbers: https://github.com/Financial-Times/o-labels/pull/104

notlee committed:
Update indicator style (animation and size) (#104)

* Update indicator style (animation and size)

- Use whole numbers for the indicator circle, to avoid subpixel
rendering issues: https://github.com/Financial-Times/o-labels/pull/103
- Update live animation as proposed by Josh from Content Innovation
and approved in the design ops meeting https://github.com/Financial-Times/o-labels/issues/101

origamiserviceuser commented:
o-labels bundle size difference from 5.1.4 to 48c29629873a3f2359e4be5078ad65e110433558
css, master: 0.65kb decrease (-0.10kb/gzip)
An insignificant difference was also found for: css (internal), css (whitelabel)

origamiserviceuser commented:
:tada: This PR is included in version v5.1.5 :tada:
